middleware.ts

```diff
@@ -1,12 +1,32 @@
-import { createMiddlewareClient } from '@supabase/auth-helpers-nextjs'
-import { NextResponse } from 'next/server'
+import { type NextRequest } from 'next/server';
+import { createClient } from '@/utils/supabase/middleware';
 
-import type { NextRequest } from 'next/server'
-import type { Database } from '@/types_db'
+export async function middleware(request: NextRequest) { 
+  const { supabase, response } = createClient(request);
+  const { error } = await supabase.auth.getSession();
+  
+  if (error?.message.match("Invalid Refresh Token")) {
```

Comment:
What is this block used for? It shouldn't be necessary, as expired sessions that fail to refresh should be automatically removed already. Another reason we probably don't want to do this, is the name of the cookie can be overwritten - if the user specifies this, it wouldn't necessarily start with `sb-`.

Reply:
> What is this block used for? It shouldn't be necessary, as expired sessions that fail to refresh should be automatically removed already. Another reason we probably don't want to do this, is the name of the cookie can be overwritten - if the user specifies this, it wouldn't necessarily start with `sb-`.

Unfortunately there is currently an open Supabase issue where if you're logged in with a cookie for a deleted user, you get an invalid refresh token error. It only happens in local development, and this code block shouldn't ever get triggered in production because Supabase usually handles this error silently by simply signing out the user. I mainly added this as a stop gap because I kept running into the issue as I worked on the PR. I have two different Supabase projects that I test on local host, so I repeatedly found myself logged in with cookies for the wrong project. See here:

https://github.com/supabase/auth-helpers/issues/436